File: src/main/java/jp/co/sunarch/apps/idprovider/model/oauth2/refresh/RefreshTokenFlow.java
package jp.co.sunarch.apps.idprovider.model.oauth2.refresh;

import jp.co.sunarch.apps.idprovider.model.oauth2.EndpointRequestFactory;
import jp.co.sunarch.apps.idprovider.model.oauth2.TokenEndpoint;
import jp.co.sunarch.apps.idprovider.model.oauth2.AccessTokenResponse;

/**
 * @author takeshi
 */
public class RefreshTokenFlow implements TokenEndpoint<RefreshTokenAccessTokenRequest, AccessTokenResponse> {

  @Override
  public EndpointRequestFactory<RefreshTokenAccessTokenRequest> tokenRequestFactory() {
    return null;
  }

  @Override
  public AccessTokenResponse createToken(RefreshTokenAccessTokenRequest request) {
    return null;
  }

}

File: src/main/kotlin/jp/co/sunarch/apps/idprovider/model/oauth2/refresh/RefreshTokenFlow.kt
package jp.co.sunarch.apps.idprovider.model.oauth2.refresh

import jp.co.sunarch.apps.idprovider.model.oauth2.EndpointRequestFactory
import jp.co.sunarch.apps.idprovider.model.oauth2.TokenEndpoint
import jp.co.sunarch.apps.idprovider.model.oauth2.AccessTokenResponse

/**
 * @author takeshi
 */
class RefreshTokenFlow : TokenEndpoint<RefreshTokenAccessTokenRequest, AccessTokenResponse> {

  override fun tokenRequestFactory(): EndpointRequestFactory<RefreshTokenAccessTokenRequest> {
    TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
  }

  override fun createToken(request: RefreshTokenAccessTokenRequest): AccessTokenResponse {
    TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
  }

}
